File: src/engine/renderer/buffers/BufferElement.java
package engine.renderer.buffers;

import engine.shaders.ShaderDataType;

public class BufferElement {
    public String name;
    public ShaderDataType type;
    public int size;
    public int offset;
    public boolean normalized;

    public BufferElement(String name, ShaderDataType type) {
        this(name, type, false);
    }

    public BufferElement(String name, ShaderDataType type, boolean normalized) {
        this.name = name;
        this.type = type;
        this.size = type.getSize();
        this.offset = 0;
        this.normalized = normalized;
    }
}

File: Lamp/Rendering/Buffers/BufferElement.cs
using Lamp.Shaders;

namespace Lamp.Rendering.Buffers
{
    public struct BufferElement
    {
        public string name;
        public ShaderDataType type;
        public int size;
        public int offset;
        public bool normalized;

        public BufferElement(string name, ShaderDataType type) : this(name, type, false) {}

        public BufferElement(string name, ShaderDataType type, bool normalized)
        {
            this.name = name;
            this.type = type;
            size = type.GetSize();
            offset = 0;
            this.normalized = normalized;
        }
    }
}
